chain/rosetta-rpc/src/lib.rs

```diff
@@ -383,13 +384,42 @@ async fn account_balance(
     } else {
         account_balances.liquid
     };
-
+    let public_keys = match account_identifier.metadata {
+        None => {
+            return Ok(Json(models::AccountBalanceResponse {
+                block_identifier: models::BlockIdentifier {
+                    hash: block_hash.to_base(),
+                    index: block_height.try_into().unwrap(),
+                },
+                balances: vec![models::Amount::from_yoctonear(balance)],
+                metadata: None,
+            }))
+        }
+        Some(metadata) => metadata,
+    };
+    let signer_public_access_key = public_keys.into_iter().next().ok_or_else(|| {
+        errors::ErrorKind::InvalidInput("exactly one public key is expected".to_string())
+    })?;
+    let (_block_hash, _block_height, access_key) = crate::utils::query_access_key(
+        near_primitives::types::BlockReference::latest(),
+        account_id_for_access_key.into(),
+        (&signer_public_access_key).try_into().map_err(|err| {
+            errors::ErrorKind::InvalidInput(format!(
+                "public key could not be parsed due to: {:?}",
+                err
+            ))
+        })?,
+        &view_client_addr,
+    )
+    .await?;
+    let nonce = access_key.nonce;
```

Comment:
Could we extract getting of the nonce to separate function? It would then also allow:
```rust
let metadata = account_identifier.metadata.map(|metadata| models::PublicKeyNonce { get_nonce(metadata) });
```
and getting a single return.

Reply:
Ah, actually await might complicate this, but even then we would have:

```rust
let metadata = if let Some(metadata) = account_identifier.metadata {
    Some(get_nonce(metadata, other args).await?)
} else {
    None
};
```

which would shorten this function and consolidate to a single return
showing that we always return the same block_identifier and balances.